wsynphot/get_filter_data.py

```diff
@@ -0,0 +1,78 @@
+import numpy as np
+import pandas as pd
+import requests
+
+from astropy.io.votable import parse_single_table
+# The VOTables fetched from SVO contain only single table element, thus parse_single_table
+
+SVO_MAIN_URL = 'http://svo2.cab.inta-csic.es/theory/fps/fps.php'
+
+# A utility fuction to convert byte strings to normal literal strings
+def byte_to_literal_strings(dataframe):
+    # Select the str columns:
+    str_df = dataframe.select_dtypes([np.object])
+    
+    # Convert all of them into unicode strings
+    str_df = str_df.stack().str.decode('utf-8').unstack()
+    
+    # Swap out converted cols with the original df cols
+    for col in str_df:
+        dataframe[col] = str_df[col]
+    return dataframe
+
+
+def get_filter_index():
+    # To get master list, fetch all filters from SVO of wavelength between 0 & very large no.
+    query = {'WavelengthEff_min': 0,
```

Comment:
make these keyword arguments and use units. I believe they are in angstrom - so specify 0 angstrom and inf angstrom as min and max. 

like this `getindex(wavelength_eff_min=0 * u.angstrom, ...)`

then convert them to angstrom by `from astropy import units as u; wavelength_eff_min = u.Quantity(wavelength_eff_min, u.angstrom)`

Reply:
No actually they are just parameters to be used to construct query string by requests - you can find more details [here](https://2.python-requests.org/en/master/user/quickstart/#passing-parameters-in-urls). The SVO HTTP queries specify parameters simply as numbers, no units.

Reply:
yes, but they have implied units - which are given in angstrom. ah - you need to convert them to int so do `int(wavelength_eff_min.value)` after converting them to angstrom

Reply:
OK I'll do that. I overlooked that keyword argument line, earlier!